[master] Scripts/Pandaria: Gate of the Setting Sun

Infamous-devel opened the pull request:
**Changes proposed**:
- Gate of the Setting Sun Instance Script
- Creatures
- Commander Rimok
- Raigonn
- Saboteur Kiptilak
- Striker Gadok 

**Target branch**:  6x

**Known issues and TODO list**:
- [x] C++ Scripts
- [x] ScriptNames SQL
- [x] Spawns (@LuigiElleBalotta #17018)

Infamous-devel committed:
Gate of the Setting Sun
+ C++ Scripts
+ ScriptNames SQL

Infamous-devel committed:
Added Copyright

Aokromes commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_striker_gadok.cpp:
```diff
@@ -15,7 +14,6 @@
  * You should have received a copy of the GNU General Public License along
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
-
```
don't remove this

Infamous-devel committed:
Added Script Loader

soulfrost commented:
![image](https://cloud.githubusercontent.com/assets/2207154/14708408/571e7cc8-07d3-11e6-8793-72ede86d48c2.png)
NICE COPY PASTE man, can u teach me master how to do this?

Aokromes commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_saboteur_kiptilak.cpp:
```diff
@@ -15,7 +14,6 @@
  * You should have received a copy of the GNU General Public License along
  * with this program. If not, see <http://www.gnu.org/licenses/>.
  */
-
```
don't remove this.

Infamous-devel commented:
@soulfrost As I told in my other PR. I take code from other emus based on TC and changed the necesary for full compatibility with the latest updates of TrinityCore.

PD: In the other PR people like you did that types of comment. I only want to help the development of that emulator. But people like you make that work harder and sometimes I guess that you forget that I'm doing this for helping. Is the last comment of that Type, if I receive another I'm not going to make more pull request or help this community.

Infamous-devel committed:
Added blank line between copy & code.

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_commander_rimok.cpp:
```diff
@@ -0,0 +1,393 @@
+    TALK_SLAY,
+};
+
+enum eEvents
```
eEvents -> Events

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_commander_rimok.cpp:
```diff
@@ -0,0 +1,393 @@
+    SPELL_BOMBARD                   = 120559
+};
+
+enum eTalks
```
eTalks -> Yells

soulfrost commented:
"and changed the necesary for full compatibility with the latest updates of TrinityCore."

bullshit, i see 100% code compability with copypasted - with hacks, wrong times, wrong events ( fe which was reworked on 5.2.x ) and other shit... i dont see here any compability with TC code standarts at least

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_commander_rimok.cpp:
```diff
@@ -0,0 +1,393 @@
+#include "ScriptedCreature.h"
+#include "gate_setting_sun.h"
+
+enum eSpells
```
eSpells -> Spells

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_commander_rimok.cpp:
```diff
@@ -0,0 +1,393 @@
+                instance = creature->GetInstanceScript();
+            }
+
+            InstanceScript* instance;
```
not needed in BossAI, it's already inherited

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_commander_rimok.cpp:
```diff
@@ -0,0 +1,393 @@
+                if (me->HasUnitState(UNIT_STATE_CASTING))
+                    return;
+
+                switch(events.ExecuteEvent())
```
add space between switch and (

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_commander_rimok.cpp:
```diff
@@ -0,0 +1,393 @@
+
+        CreatureAI* GetAI(Creature* creature) const
+        {
+            return new boss_commander_rimokAI(creature);
```
use GetInstanceAI

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,664 @@
+#include "Vehicle.h"
+#include "Spline.h"
+
+enum eSpells
```
the same codestyle changes as in previous file

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,664 @@
+                pInstance = creature->GetInstanceScript();
+            }
+
+            InstanceScript* pInstance;
```
not needed, use inherited "instance"

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,664 @@
+                    {
+                        weakSpot->EnterVehicle(me, 1);
+
+                        if (pInstance)
```
not needed when you use inherited instance and GetInstanceAI

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,664 @@
+                if (me->GetDistance(who) > 30.0f)
+                    return;
+
+                if (pInstance)
```
the same once more

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,664 @@
+            uint8  eventChargeProgress;
+            uint32 eventChargeTimer;
+
+            uint8 Phase;
```
maybe consider of using EventMap phases

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,664 @@
+
+            void JustSummoned(Creature* summoned)
+            {
+                summons.Summon(summoned);
```
I think this is already of part of parent class BossAI, should not be needed, check ScriptedCreature.h/cpp

Infamous-devel commented:
@soulfrost Maybe in this Instance isn't any change but I have been fixing and cchanging code for free of error compiling. And what if is copy/paste this is OPEN SOURCE.

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,664 @@
+
+            void RemoveWeakSpotPassengers()
+            {
+                if (Creature* weakPoint = pInstance->GetCreature(pInstance->GetData64(NPC_WEAK_SPOT)))
```
use ObjectAccessor::GetCreature(*me, instance->GetData64(NPC_WEAK_SPOT) )

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,664 @@
+
+            void DoEventCharge()
+            {
+                if (!pInstance)
```
the same as above

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,664 @@
+                if (!me->SelectNearestPlayer(25.0f))
+                    return false;
+
+                if (pInstance)
```
once more

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,664 @@
+
+            void UpdateAI(uint32 diff)
+            {
+                if (!pInstance)
```
once more

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,664 @@
+
+            void JustDied(Unit* /*killer*/)
+            {
+                events.Reset();
```
do you really need reset eventmap?

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,664 @@
+                events.Reset();
+                if (instance)
+                {
+                    instance->SetBossState(DATA_RAIGONN, DONE);
```
BossAI::JustDied should does all these lines

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,664 @@
+
+        CreatureAI* GetAI(Creature* creature) const
+        {
+            return new boss_raigonnAI(creature);
```
use InstanceAI here as well

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_saboteur_kiptilak.cpp:
```diff
@@ -0,0 +1,367 @@
+#include "ScriptedCreature.h"
+#include "gate_setting_sun.h"
+
+enum eTalks
```
again codestyle changes, don't use "e" prefix for enums

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_saboteur_kiptilak.cpp:
```diff
@@ -0,0 +1,367 @@
+                instance = creature->GetInstanceScript();
+            }
+
+            InstanceScript* instance;
```
again use inherited one

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_saboteur_kiptilak.cpp:
```diff
@@ -0,0 +1,367 @@
+                std::list<Creature*> munitionList;
+                GetCreatureListWithEntryInGrid(munitionList, me, NPC_STABLE_MUNITION, 100.0f);
+
+                for (auto itr: munitionList)
```
consider using reference

Shauren replied:
Simple type here (pointer) - dont, also don't use 'auto' for non-iterator types

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_saboteur_kiptilak.cpp:
```diff
@@ -0,0 +1,367 @@
+
+        CreatureAI* GetAI(Creature* creature) const
+        {
+            return new boss_saboteur_kiptilakAI(creature);
```
GetInstanceAI once more

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_striker_gadok.cpp:
```diff
@@ -0,0 +1,752 @@
+        {
+            boss_striker_gadokAI(Creature* creature) : BossAI(creature, DATA_GADOK)
+            {
+                instance = creature->GetInstanceScript();
```
again

MitchesD replied:
remove that line at ALL, you already inherited it with value, you can't assign new

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_striker_gadok.cpp:
```diff
@@ -0,0 +1,752 @@
+    STRAF_30      = 2
+};
+
+#define MAX_DISRUPTOR   5
```
use rather Misc enum instead of define if you could

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_striker_gadok.cpp:
```diff
@@ -0,0 +1,752 @@
+                DoSpawnKrikThik();
+
+                if (instance)
+                    instance->SetData(DATA_GADOK, PHASE_MAIN);
```
sure? _Reset() changes boss state of DATA_GADOK, is that intended or some leftover?

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_striker_gadok.cpp:
```diff
@@ -0,0 +1,752 @@
+
+            void JustReachedHome()
+            {
+                if (instance)
```
remove

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_striker_gadok.cpp:
```diff
@@ -0,0 +1,752 @@
+                        else if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 0.0f, true))
+                            me->AI()->AttackStart(target);
+
+                        if (instance)
```
remove

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/gate_setting_sun.h:
```diff
@@ -0,0 +1,116 @@
+    GO_PORTAL_TEMP_GADOK        = 400002
+};
+
+enum eSettingSun
```
again "e"

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,399 @@
+        uint32 cinematicTimer;
+        uint8 cinematicEventProgress;
+
+        std::list<uint64> bombarderGuids;
```
maybe GuidSet?

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,399 @@
+
+    struct instance_gate_setting_sun_InstanceMapScript : public InstanceScript
+    {
+        uint64 kiptilakGuid;
```
GUID isn't uint64, but ObjectGuid

Lordron replied:
take a look at https://github.com/TrinityCore/TrinityCore/commit/b08dcb1d99fe91b7dc1cb3168a34d722f9ea5505

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,399 @@
+        void OnPlayerEnter(Player* player)
+        {
+            if (GetData(DATA_BRASIER_CLICKED) == NOT_STARTED)
+                player->SetPhaseMask(1, true);
```
there is no phaseMask on 6x, please don't use it, there are just ancient methods in core

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,399 @@
+            switch (creature->GetEntry())
+            {
+                case NPC_KRITHUK_BOMBARDER:
+                    for (std::list<uint64>::iterator it = bombarderGuids.begin(); it != bombarderGuids.end(); ++it)
```
as I said, GUIDs are type of ObjectGuid, also there you can use range-based for loop

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,399 @@
+                case DATA_KIPTILAK:
+                {
+                    if (state == DONE)
+                        for (std::list<uint64>::const_iterator itr = mantidBombsGUIDs.begin(); itr != mantidBombsGUIDs.end(); ++itr)
```
the same here

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,399 @@
+                    if (state == DONE)
+                        for (std::list<uint64>::const_iterator itr = mantidBombsGUIDs.begin(); itr != mantidBombsGUIDs.end(); ++itr)
+                            if (GameObject* bomb = instance->GetGameObject(GetObjectGuid(*itr)))
+                                bomb->SetPhaseMask(32768, true); // Set Invisible
```
again, use Phases instead of phaseMask

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,399 @@
+                case DATA_GADOK:
+                {
+                    if (GameObject* portal = instance->GetGameObject(GetObjectGuid(portalTempGadokGuid)))
+                        portal->SetPhaseMask(state == IN_PROGRESS ? 4 : 3, true);
```
here as well

Infamous-devel committed:
Fixes:
+ Code Style
+ Rename of Enums
+ GetAI to GetInstanceAI

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,399 @@
+                {
+                    uint8 generatorsCount = 0;
+
+                    for (std::list<uint64>::const_iterator itr = secondaryDoorGUIDs.begin(); itr != secondaryDoorGUIDs.end(); ++itr)
```
there are multiple cases, won't comment all, just change them all

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,399 @@
+                            std::list<Creature*> defensorList;
+                            GetCreatureListWithEntryInGrid(defensorList, trigger, 65337, 20.0f);
+
+                            trigger->CastSpell(trigger, 115456); // Explosion
```
don't use magic numbers

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,399 @@
+                        if (Player* player = it->GetSource())
+                        {
+                            player->SendCinematicStart(CINEMATIC_SETTING_SUN);
+                            player->SetPhaseMask(2, true);
```
again phasemask

MitchesD commented:
@Infamous-devel  I commented a few lines, there are many wrong things, I didn't comment all of them, but check for cases I commented, like Guids, phaseMasks, boss states, inherited methods/members etc.

Infamous-devel committed:
+ Fixed Code Style.
+ Replace Old/Unused Functions.
+ Replace Magic Number.
+ Tried to do all MitchesD said. (If I know how to).
+ Added missed sql area_trigger script.

Infamous-devel committed:
Remove inherited

Keader commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,648 @@
+                    DoMeleeAttackIfReady();
+            }
+
+            void JustDied(Unit* killer)
```
killer is used?
if not, Travis complain of this

Infamous-devel committed:
Replace "auto" varibles with real type.

Infamous-devel committed:
Clean Gadok Reset()

Infamous-devel committed:
Comment unused variable.

Keader commented on sql/updates/world/6.x/2016_04_21_00_world.sql:
```diff
@@ -0,0 +1,43 @@
+UPDATE creature_template SET ScriptName='npc_krikthik_disruptor' WHERE entry=59794;
+UPDATE creature_template SET ScriptName='npc_flak_cannon' WHERE entry=60421;
+INSERT INTO spell_script_names (spell_id,ScriptName) values (116298,'spell_gadok_strafing');
+INSERT INTO spell_script_names (spell_id,ScriptName) values (106933,'spell_prey_time');
```
add 1 new line here

Keader commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_commander_rimok.cpp:
```diff
@@ -0,0 +1,389 @@
+
+                if (checkTimer <= diff)
+                {
+                    me->CastSpell(me, SPELL_BOMBARD, false);
```
false is default value, so you can remove...
like:  me->CastSpell(me, SPELL_BOMBARD);

MitchesD replied:
actually better is DoCast

Infamous-devel commented:
@Keader I comment that variable. Travis exit with error code when warning? That is a bit rare... In local I can compile with warnings without problems.

Keader commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,648 @@
+                switch (eventChargeProgress)
+                {
+                    case 0:
+                        me->CastSpell(me, SPELL_BATTERING_HEADBUTT_EMOTE, false);
```
same here

Keader commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,648 @@
+                    }
+                    case EVENT_STOMP:
+                    {
+                        me->CastSpell(me, SPELL_STOMP, false);
```
here too

Keader commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,648 @@
+                if (engulfingTimer <= diff)
+                {
+                    if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 1))
+                        me->CastSpell(target, SPELL_ENGULFING_WINDS, false);
```
here

Keader commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_raigonn.cpp:
```diff
@@ -0,0 +1,648 @@
+                if (swarmTimer <= diff)
+                {
+                    if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 1))
+                        me->CastSpell(target, SPELL_SCREECHING_SWARM, false);
```
here

Infamous-devel committed:
SQL added final blank line.
CastSpell without "false" argument, because is default value.

Keader commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_saboteur_kiptilak.cpp:
```diff
@@ -0,0 +1,365 @@
+            damage = 0;
+        }
+
+        void MovementInform(uint32 type, uint32 id)
```
type is unsuded, travis should complain

Keader commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_saboteur_kiptilak.cpp:
```diff
@@ -0,0 +1,365 @@
+            me->AddAura(SPELL_MUNITION_EXPLOSION_VISUAL, me);
+        }
+
+        void DamageTaken(Unit* attacker, uint32& damage)
```
type is unused, travis should complain

Infamous-devel committed:
Remove CastSpell default argument.

Infamous-devel committed:
Comment unused arguments.

Infamous-devel committed:
Comment unused arguments.

Keader commented:
Yeap, some warnings == fatal error for Travis xD

Infamous-devel committed:
Comment more unused arguments.

Keader commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_commander_rimok.cpp:
```diff
@@ -0,0 +1,389 @@
+                {
+                    case EVENT_FRENZIED_ASSAULT:
+                        if (me->GetVictim())
+                            me->CastSpell(me->GetVictim(), SPELL_FRENZIED_ASSAULT, false);
```
you can change this:
```
if (me->GetVictim())
me->CastSpell(me->GetVictim(), SPELL_FRENZIED_ASSAULT, false);
```

for this: 
```
DoCastVictim(SPELL_FRENZIED_ASSAULT);
``` 

or:
```
if (me->GetVictim())
me->CastSpell(me->GetVictim(), SPELL_FRENZIED_ASSAULT);
```

ignore my tabs lol

Keader commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_striker_gadok.cpp:
```diff
@@ -0,0 +1,749 @@
+                {
+                    case EVENT_PREY_TIME:
+                        if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 0.0f, true))
+                            me->CastSpell(target, SPELL_PREY_TIME, false);
```
false again xD

Keader commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_striker_gadok.cpp:
```diff
@@ -0,0 +1,749 @@
+                    case EVENT_IMPALING_STRIKE:
+                        if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 0.0f, true))
+                            me->CastSpell(target, SPELL_IMPALING_STRIKE, false);
+
```
here too

Keader commented:
Travis: https://travis-ci.org/TrinityCore/TrinityCore/builds/124754610

Exodius commented on sql/updates/world/6.x/2016_04_21_00_world.sql:
```diff
@@ -0,0 +1,43 @@
+-- Gate of the Setting Sun
+UPDATE instance_template SET script='instance_gate_setting_sun' WHERE map=962;
+UPDATE creature_template SET ScriptName='mob_serpent_spine_defender' WHERE entry=58146;
```
For future reference, when doing the cleanup of the scripts, be sure to change `mob_` prefixes to `npc_`.
Overall, great effort. I'm glad you decided to separate one massive PR into smaller ones, one per instance. +1

Infamous-devel committed:
Fixes:
+ "read-only variable is not assignable"
+ "hidden overloaded virtual function 'AreaTriggerScript::OnTrigger'
declared"

Infamous-devel committed:
+ Cast Spell delete default argument "false".
+ Rename 'mob_serpent_spine_defender' to 'npc_serpent_spine_defender'

Infamous-devel commented:
Travis error installing pre-requisites: The command "sudo apt-get -qq install libboost1.55-dev libboost-thread1.55-dev libboost-filesystem1.55-dev libboost-system1.55-dev libboost-program-options1.55-dev libboost-iostreams1.55-dev libboost-regex1.55-dev" failed and exited with 100 during .

Shauren commented:
Ingore travis errors like that, servers are probably under heavy load thanks to new ubuntu release yesterday

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_commander_rimok.cpp:
```diff
@@ -0,0 +1,386 @@
+
+        CreatureAI* GetInstanceAI(Creature* creature) const
+        {
+            return new boss_commander_rimokAI(creature);
```
as I said, use GetInstanceAI like: return GetInstanceAI<boss_commander_rimokAII>(creature);

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_saboteur_kiptilak.cpp:
```diff
@@ -0,0 +1,364 @@
+
+        CreatureAI* GetInstanceAI(Creature* creature) const
+        {
+            return new boss_saboteur_kiptilakAI(creature);
```
return GetInstanceAI<boss_saboteur_kiptilakAI>(creature);

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_saboteur_kiptilak.cpp:
```diff
@@ -0,0 +1,364 @@
+            }
+        };
+
+        CreatureAI* GetInstanceAI(Creature* creature) const
```
not sure why you are using GetInstanceAI here, you should keep ```CreatureAI* GetAI(Creature* creature) const override```

Lordron commented:
@Infamous-devel take a look at https://github.com/TrinityCore/TrinityCore/commit/b08dcb1d99fe91b7dc1cb3168a34d722f9ea5505

Keader commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_commander_rimok.cpp:
```diff
@@ -0,0 +1,386 @@
+                switch (events.ExecuteEvent())
+                {
+                    case EVENT_FRENZIED_ASSAULT:
+                        DoCastVictim(SPELL_FRENZIED_ASSAULT, false);
```
false is default value for triggered in  DoCastVictim too

Infamous-devel committed:
+ Fix GetAI returns.
+ Fix compile warnings.

Infamous-devel committed:
+ Spawns Added (by @LuigiElleBalotta)

Infamous-devel committed:
+ Fix wrong override

Infamous-devel committed:
Merge remote-tracking branch 'upstream/6.x' into gotss

Infamous-devel committed:
+ Fix unused argument warning.
+ Fix "no viable overloaded".

Infamous-devel committed:
+ Fix type error.
+ Remove unused argument.

Infamous-devel committed:
+ Clean unused argument.
+ Override OnTrigger method.

Infamous-devel committed:
+ Comment unused argument.

Infamous-devel committed:
+ Comment error giving function call. (Not sure how should it work).

Infamous-devel commented:
Finally compiles without errors/warnings :+1:

LuigiElleBalotta commented:
Why it isn't merged yet?

tkrokli commented:
Probably because it's not properly reviewed, validated or labeled as [Feedback-PatchFix] yet.

Infamous-devel commented:
Waiting news!

horn commented:
A lot of overrides missing, EventMap should be used for timers instead of <code>variable -= diff;</code>, also what's wrong with all that code commented out?

ccrs commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,404 @@
+        instance_gate_setting_sun_InstanceMapScript(Map* map) : InstanceScript(map)
+        {}
+
+        void Initialize()
```
never use initialize, place this inside the constructor

ccrs commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,404 @@
+        uint32 cinematicTimer;
+        uint8 cinematicEventProgress;
+
+        std::list<ObjectGuid> bombarderGuids;
```
GuidList

ccrs commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,404 @@
+        std::list<ObjectGuid> artilleryGUIDs;
+        std::list<ObjectGuid> secondaryDoorGUIDs;
+
+        uint32 dataStorage[MAX_DATA];
```
unneded

ccrs commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,404 @@
+        void OnPlayerEnter(Player* player)
+        {
+            if (GetData(DATA_BRASIER_CLICKED) == NOT_STARTED)
+                player->SetInPhase(1, true, true);
```
I bet this is handled with a spell

ccrs commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,404 @@
+        {
+            switch (creature->GetEntry())
+            {
+                case NPC_KIPTILAK:          kiptilakGuid    = creature->GetGUID();                  return;
```
ermmmm codestyle

tkrokli replied:
Maybe something like this instead?
```cpp
            {
                case NPC_KIPTILAK:
                    kiptilakGuid = creature->GetGUID();
                    return;
                case NPC_GADOK:
                    gadokGuid = creature->GetGUID();
                    return;
                case NPC_RIMOK:
                    rimokGuid = creature->GetGUID();
                    return;
                case NPC_RAIGONN:
                    raigonnGuid = creature->GetGUID();
                    return;
                case NPC_KRITHUK_BOMBARDER:
                    bombarderGuids.push_back(creature->GetGUID());
                    return;
                case NPC_BOMB_STALKER:
                    bombStalkerGuids.push_back(creature->GetGUID());
                    return;
                case NPC_ADD_GENERATOR:
                    rimokAddGenetarorsGUIDs.push_back(creature->GetGUID());
                    return;
                case NPC_ARTILLERY:
                    artilleryGUIDs.push_back(creature->GetGUID());
                    return;
                default:
                    return;
            }
```

ccrs commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,404 @@
+            }
+        }
+
+        virtual void OnCreatureRemove(Creature* creature)
```
remove virtual

ccrs commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,404 @@
+            return 0;
+        }
+
+        void SetData64(uint32 type, uint64 value)
```
deprecated

ccrs commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,404 @@
+            }
+        }
+
+        uint64 GetData64(uint32 type) const override 
```
deprecated

ccrs commented:
missing overrides everywhere

this wont get merged if you dont use sniffs

tkrokli commented on src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp:
```diff
@@ -0,0 +1,404 @@
+        uint64 GetData64(uint32 type) const override 
+        {
+            switch (type)
+            {
```
... if you are going to use this switch case with a different function or method, use TC codestyle:
```cpp
            {
                case NPC_KIPTILAK:
                    return kiptilakGuid.GetEntry();
                case NPC_GADOK:
                    return gadokGuid.GetEntry();
                case NPC_RIMOK:
                    return rimokGuid.GetEntry();
                case NPC_RAIGONN:
                    return raigonnGuid.GetEntry();
                case NPC_WEAK_SPOT:
                    return raigonWeakGuid.GetEntry();
                case DATA_RANDOM_BOMBARDER:
                    return Trinity::Containers::SelectRandomContainerElement(bombarderGuids).GetEntry();
                case DATA_RANDOM_BOMB_STALKER:
                    return Trinity::Containers::SelectRandomContainerElement(bombStalkerGuids).GetEntry();
            }
```

tkrokli commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_striker_gadok.cpp:
```diff
@@ -0,0 +1,750 @@
+                // In videos, Gadok follow the opposite direction of clockwise, to confirm
+                switch (move)
+                {
+                    case MOV_NORTH_SOUTH:   return MOV_WEST_EAST;
```
Codestyle:
```cpp
                {
                    case MOV_NORTH_SOUTH:
                        return MOV_WEST_EAST;
                    case MOV_SOUTH_NORTH:
                        return MOV_EAST_WEST;
                    case MOV_WEST_EAST:
                        return MOV_NORTH_SOUTH;
                    case MOV_EAST_WEST:
                        return MOV_SOUTH_NORTH;
                    default:
                        break;
                }
```

tkrokli commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_striker_gadok.cpp:
```diff
@@ -0,0 +1,750 @@
+
+                switch (id)
+                {
+                    case POINT_NORTH_START:   case POINT_SOUTH_START:   case POINT_WEST_START:   case POINT_EAST_START:
```
... more codestyle:
```cpp
                {
                    case POINT_NORTH_START:
                    case POINT_SOUTH_START:
                    case POINT_WEST_START:
                    case POINT_EAST_START:
                    case POINT_NORTH_END:
                    case POINT_SOUTH_END:
                    case POINT_WEST_END:
                    case POINT_EAST_END:
                    case POINT_NORTH_OUTSIDE:
                    case POINT_SOUTH_OUTSIDE:
                    case POINT_WEST_OUTSIDE:
                    case POINT_EAST_OUTSIDE:
                        DoStrafingEvent();
                        break;
                    default:
                        break;
                }
```

Aokromes commented:
@Infamous-devel plz read codestyle things :)

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_commander_rimok.cpp:
```diff
@@ -0,0 +1,386 @@
+                        events.ScheduleEvent(EVENT_FRENZIED_ASSAULT, urand(10000, 15000));
+                        break;
+                    case EVENT_VISCOUS_FLUID:
+                        me->CastSpell(me, 107077, true);
```
don't use magic numbers and also you should use DoCast(me, SPELL, true);

tkrokli replied:
"don't use magic numbers" = add 107077 to the `enum Spells` list:
```cpp
enum Spells
{
    // Commander Rimok
    SPELL_VISCOUS_FLUID           = 107077,
    SPELL_VISCOUS_FLUID_SUMMON    = 107078,
    SPELL_VISCOUS_FLUID_DMG_UP    = 107091,
    SPELL_VISCOUS_FLUID_DMG_DOWN  = 107122,
    SPELL_FRENZIED_ASSAULT        = 107120,

    // Add Generator
    SPELL_PERIODIC_SPAWN_SWARMER  = 115052,
    SPELL_PERIODIC_SPAWN_SABOTEUR = 116621,

    // Saboteur
    SPELL_BOMBARD                 = 120559
};
```
```cpp
                         me->DoCast(me, SPELL_VISCOUS_FLUID, true);
```
(I also decreased the `= spell_id` indentation, feel free to keep your original indentation)

tkrokli replied:
```cpp
                    case EVENT_VISCOUS_FLUID:
                        DoCast(me, SPELL_VISCOUS_FLUID_SUMMON, true);
```
Viscous Fluid enum changes:
```cpp
enum Spells
{
    // Commander Rimok
    SPELL_VISCOUS_FLUID_SUMMON      = 107077, // Effect: Summon (Viscos Fluid Stalker)
    SPELL_VISCOUS_FLUID_APPLY_AURA  = 107078, // Effect: Apply Aura: Periodically trigger spell
    SPELL_VISCOUS_FLUID_DMG_UP      = 107091, // Effect: Apply Aura: Mod Damage Done (Value: +10%  +20%)
    SPELL_VISCOUS_FLUID_DMG_DOWN    = 107122, // Effect: Apply Aura: Mod Damage Done (Value: -10%  -20%)
    SPELL_FRENZIED_ASSAULT          = 107120,

    // Add Generator
    SPELL_PERIODIC_SPAWN_SWARMER    = 115052,
    SPELL_PERIODIC_SPAWN_SABOTEUR   = 116621,

    // Saboteur
    SPELL_BOMBARD                   = 120559
};
```
Sorry for changing your enum names, I based them on the wowhead.com description:
- http://www.wowhead.com/spell=107077/viscous-fluid (Effect: Summon (Viscos Fluid Stalker))
- http://www.wowhead.com/spell=107078/viscous-fluid (Apply Aura: Periodically trigger spell)

MitchesD commented on src/server/scripts/Pandaria/GateoftheSettingSun/boss_commander_rimok.cpp:
```diff
@@ -0,0 +1,386 @@
+                        me->RemoveAurasDueToSpell(SPELL_PERIODIC_SPAWN_SABOTEUR);
+                        break;
+                    case SPECIAL: // Only one must spawn saboteurs
+                        me->CastSpell(me, SPELL_PERIODIC_SPAWN_SABOTEUR, true);
```
use DoCast() instead

tkrokli replied:
```cpp
                         me->DoCast(me, SPELL_PERIODIC_SPAWN_SABOTEUR, true);
```

Aokromes commented:
@Infamous-devel plz read codestyle things :)

Infamous-devel commented:
Will read the codestyle and review this code soon. :+1:

Infamous-devel committed:
Merge remote-tracking branch 'origin/6.x' into gotss

Infamous-devel committed:
Codestyle fixes.
More suggested fixes.

Aokromes commented:
Any oposition to merge this? if no i will merge it on 24h.

LuigiElleBalotta commented:
News?

Infamous-devel commented:
Not have much time to review and fix the PR. @Aokromes can merge if want, could fix it after.

Kittnz commented:
There should be atleast a cleanup.

Arles91 commented:
It Works?

tkrokli commented:
@Arles91 : We don't know more than what is already documented in this PR.
But it would be nice if you could test it and report back with your results.

CarlosX commented:
Work ?

joschiwald commented:
Superseded by #21924
